src/include/daos_task.h

```diff
@@ -421,14 +421,38 @@ typedef struct {
 typedef struct {
 	daos_handle_t		oh;
 	daos_handle_t		th;
+	unsigned int		flags;
+	unsigned int		shard;
```

Comment:
This data structure is also part of the public API, user might have no idea about what they are.
I think you might have to define another one in src/include/daos/object.h, for example:
 struct daos_obj_rw_opt {
     daos_obj_rw_t        rw_args;
     int                           rw_flags;
     int                           rw_shard;
}

then define a separate task for it, although you can share the internal functions

Reply:
What we export is the fetch/update API:

int     
daos_obj_fetch(daos_handle_t oh, daos_handle_t th, daos_key_t *dkey,
               unsigned int nr, daos_iod_t *iods, d_sg_list_t *sgls,
               daos_iom_t *maps, daos_event_t *ev)

It does NOT contain the daos_obj_fetch_t data structure. The daos_obj_fetch_t is just used inside daos client stack.

Reply:
I will add a new define a new type task for that.